api/v1beta1/azuremachine_validation_test.go

```diff
@@ -68,13 +68,13 @@ func TestAzureMachine_ValidateSSHKey(t *testing.T) {
 	}
 }
 
-func generateSSHPublicKey(b64Enconded bool) string {
+func generateSSHPublicKey(b64Enconded bool) *string {
 	privateKey, _ := rsa.GenerateKey(rand.Reader, 2048)
 	publicRsaKey, _ := ssh.NewPublicKey(&privateKey.PublicKey)
 	if b64Enconded {
-		return base64.StdEncoding.EncodeToString(ssh.MarshalAuthorizedKey(publicRsaKey))
+		return pointer.String(base64.StdEncoding.EncodeToString(ssh.MarshalAuthorizedKey(publicRsaKey)))
 	}
-	return string(ssh.MarshalAuthorizedKey(publicRsaKey))
+	return pointer.String(string(ssh.MarshalAuthorizedKey(publicRsaKey)))
```

Comment:
Can we instead return the string from the `generateSSHPublicKey` function so that the return variable doesn't _"escape"_ the function?
We can convert the returned string to a pointer as needed.

Reply:
Sure, it should be done like you said.

Reply:
Done.